hoomd/mesh.py

```diff
@@ -86,46 +95,48 @@ def _attach_hook(self):
                 self._cpp_obj.setCommunicator(
                     self._simulation._system_communicator)
 
-    @log(category='sequence')
-    def triangles(self):
-        """((*N*, 3) `numpy.ndarray` of ``uint32``): Mesh triangulation.
+    def _ensure_same_size(self, triangulation):
+        if len(triangulation["triangles"]) != len(triangulation["type_ids"]):
+            raise ValueError(
+                "Number of type_ids do not match number of triangles.")
 
-        A list of triplets of particle tags which encodes the
-        triangulation of the mesh structure.
-        """
-        if self._attached:
-            return self._cpp_obj.getTriangleData().group
-        return self._triangles
+    def _setattr_param(self, attr, value):
+        if attr == "triangulation":
+            self._ensure_same_size(value)
+            self._set_triangulation(value)
+            return
+        super()._setattr_param(attr, value)
 
-    @triangles.setter
-    def triangles(self, triag):
-        if self._attached:
+    def _getattr_param(self, attr):
+        if attr == "triangulation":
+            return self._get_triangulation()
+        return super()._getattr_param(attr)
 
-            if len(triag) != len(self._type_ids):
-                raise ValueError(
-                    "Number of type_ids do not match number of triangles.")
-            else:
-                self._cpp_obj.setTypes(list(self._param_dict['types']))
+    def _get_triangulation(self):
+        if self._attached:
+            return dict(type_ids=self.type_ids, triangles=self.triangles)
+        return self._param_dict._dict["triangulation"]
```

Comment:
Same as below. This should be able to be handled by the ParameterDict automatically.